### DISCUSSION TITLE
Show line number at cursor position

### DISCUSSION THREAD

**Discussion by OslayViera** [arkypita/LaserGRBL#1484] [Ideas]
I think there should be a way to know the exact line where the cursor is in the image area, sometimes I have had to restart the project on a specific line and I have no way because the line number is only displayed when a work without closing it after a stop, if for some reason, for example, the PC shuts down, there is no way to get to the point where it stopped, even if the machine can be put in that position. It would be ideal that when moving the cursor somewhere it says (at least approximate) the number of the line it is on or that the job can be started from the cursor position as an option.

**Comment by arkypita** [comment]
> way to know the exact line where the cursor is in the image area

Yes, I could add a similar thing, but keep in mind that it would be a somewhat rough feature. Keep in mind that this is not even simple / possible to do in all cases. For example with vector file is quite impossible.

Actually with a line 2 line job you can do it by yourself: save the file as an .nc file, open it with a text editor, locate a line where the X / Y coordinates are close to your stop point, and delete everything there is before that line.

Make sure there is an M3 / M4 instruction on the first line to turn on the laser (add it) and save the file.

> if for some reason, for example, the PC shuts down

In the past I try to add some feature to save at runtime the current line progress info on a file on disk, But if PC crash or suddenly power down, last written files usually get corrupted.

**Reply by OslayViera** [reply]
LaserGRBL is cool as it is, it's important to say that. I have used many other programs and there is none so light and more for what I use it, I do not have a laser burner but a nichrome burner and in the latter, speed is essential. In GRBLcontroller for example it is possible to increase the preview to the point of seeing the lines and there if I think it would be easier to deduce which line the cursor is on. The point is that according to my burner, an approximate line number of ± 20 lines would serve the problem is that at the moment it is impossible for me to obtain that number because not even in the sessionlog the stop line is stored and there is no possibility of starting from the cursor position (which would also work).
Another thing that would be very useful for my specific burner is to make a section in Settings> GCode for when the AbortFile command is given that is different from Footer, because in my case when I stop a running job, even if the nichrome remains without supply it is still hot and creates an ugly hole in the wood. But hey, these are just suggestions that I suppose would be more useful to me than most of the people who use LaserGRBL although I know of a fairly large Russian community that uses your program. Well nothing, greetings from Cuba and sorry if there are translation errors, I speak little English
